File: AndroidApp/app/src/main/java/com/comentarista/futebol/data/remote/dto/RelatorioDtos.kt
package com.comentarista.futebol.data.remote.dto

import kotlinx.serialization.Serializable

// Espelha ControleFutebolWeb/Models/Api/RelatorioDtos.cs (GET api/v1/relatorios/resumo).

@Serializable
data class RelatorioResumoDto(
    val temporada: Int? = null,
    val temporadasDisponiveis: List<Int> = emptyList(),
    val competicoes: List<CompeticaoRefDto> = emptyList(),
    val exibirSelecao: Boolean = false,

    val totalJogos: Int = 0,
    val totalGols: Int = 0,
    val totalGolsContra: Int = 0,
    val totalCartaoAmarelo: Int = 0,
    val totalCartaoVermelho: Int = 0,

    val rankingNotas: List<RankingNotaDto> = emptyList(),
    val rankingGoleiros: List<RankingNotaDto> = emptyList(),
    val rankingDefensores: List<RankingNotaDto> = emptyList(),
    val rankingMeias: List<RankingNotaDto> = emptyList(),
    val rankingAtacantes: List<RankingNotaDto> = emptyList(),

    val artilheiros: List<JogadorValorDto> = emptyList(),
    val assistencias: List<JogadorValorDto> = emptyList(),
    val maisPartidas: List<JogadorValorDto> = emptyList(),
    val maisCartoesAmarelos: List<JogadorValorDto> = emptyList(),
    val maisCartoesVermelhos: List<JogadorValorDto> = emptyList(),

    val timesAproveitamento: List<TimeEstatisticaDto> = emptyList(),
    val timesMaisPontos: List<TimeEstatisticaDto> = emptyList(),
    val timesGols: List<TimeEstatisticaDto> = emptyList(),
    val timesMenosGolsSofridos: List<TimeEstatisticaDto> = emptyList(),

    val mediasPorPosicao: List<MediaPosicaoDto> = emptyList()
)

@Serializable
data class RankingNotaDto(
    val jogadorId: Int,
    val nome: String,
    val nomeExibicao: String = "",
    val posicao: String = "",
    val timeNome: String? = null,
    val fotoUrl: String? = null,
    val notaFinal: Double,
    val notaLabel: String = "",
    val notaColor: String = "",
    val partidas: Int = 0,
    val vitorias: Int = 0,
    val derrotas: Int = 0,
    val empates: Int = 0
)

@Serializable
data class JogadorValorDto(
    val jogadorId: Int,
    val nome: String,
    val nomeExibicao: String = "",
    val posicao: String = "",
    val timeNome: String? = null,
    val fotoUrl: String? = null,
    val valor: Int,
    val detalhe: String = ""
)

@Serializable
data class TimeEstatisticaDto(
    val timeId: Int,
    val nome: String,
    val escudoUrl: String? = null,
    val jogos: Int = 0,
    val vitorias: Int = 0,
    val empates: Int = 0,
    val derrotas: Int = 0,
    val golsPro: Int = 0,
    val golsContra: Int = 0,
    val saldoGols: Int = 0,
    val pontos: Int = 0,
    val aproveitamento: Double = 0.0
)

@Serializable
data class MediaPosicaoDto(
    val posicao: String,
    val media: Double,
    val totalJogadores: Int
)

File: ControleFutebolWeb/Models/Api/RelatorioDtos.cs
namespace ControleFutebolWeb.Models.Api
{
    // Resumo dos relatórios para o app mobile: os rankings mais usados da página
    // /Relatorios. Rankings menos usados (estatísticas por competição, match-up,
    // stats de jogo por time) ficam de fora por enquanto.
    public class RelatorioResumoDto
    {
        public int? Temporada { get; set; }
        public List<int> TemporadasDisponiveis { get; set; } = new();
        // Todas as competições (para o filtro do app), na mesma ordem da web
        // (top tier do usuário primeiro)
        public List<CompeticaoRefDto> Competicoes { get; set; } = new();
        public bool ExibirSelecao { get; set; }

        public int TotalJogos { get; set; }
        public int TotalGols { get; set; }
        public int TotalGolsContra { get; set; }
        public int TotalCartaoAmarelo { get; set; }
        public int TotalCartaoVermelho { get; set; }

        public List<RankingNotaDto> RankingNotas { get; set; } = new();
        public List<RankingNotaDto> RankingGoleiros { get; set; } = new();
        public List<RankingNotaDto> RankingDefensores { get; set; } = new();
        public List<RankingNotaDto> RankingMeias { get; set; } = new();
        public List<RankingNotaDto> RankingAtacantes { get; set; } = new();

        public List<JogadorValorDto> Artilheiros { get; set; } = new();
        public List<JogadorValorDto> Assistencias { get; set; } = new();
        public List<JogadorValorDto> MaisPartidas { get; set; } = new();
        public List<JogadorValorDto> MaisCartoesAmarelos { get; set; } = new();
        public List<JogadorValorDto> MaisCartoesVermelhos { get; set; } = new();

        public List<TimeEstatisticaDto> TimesAproveitamento { get; set; } = new();
        public List<TimeEstatisticaDto> TimesMaisPontos { get; set; } = new();
        public List<TimeEstatisticaDto> TimesGols { get; set; } = new();
        public List<TimeEstatisticaDto> TimesMenosGolsSofridos { get; set; } = new();

        public List<MediaPosicaoDto> MediasPorPosicao { get; set; } = new();
    }

    public class RankingNotaDto
    {
        public int JogadorId { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string NomeExibicao { get; set; } = string.Empty;
        public string Posicao { get; set; } = string.Empty;
        public string? TimeNome { get; set; }
        public string? FotoUrl { get; set; }
        public double NotaFinal { get; set; }
        public string NotaLabel { get; set; } = string.Empty;
        public string NotaColor { get; set; } = string.Empty;
        public int Partidas { get; set; }
        public int Vitorias { get; set; }
        public int Derrotas { get; set; }
        public int Empates { get; set; }
    }

    public class JogadorValorDto
    {
        public int JogadorId { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string NomeExibicao { get; set; } = string.Empty;
        public string Posicao { get; set; } = string.Empty;
        public string? TimeNome { get; set; }
        public string? FotoUrl { get; set; }
        public int Valor { get; set; }
        public string Detalhe { get; set; } = string.Empty;
    }

    public class TimeEstatisticaDto
    {
        public int TimeId { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string? EscudoUrl { get; set; }
        public int Jogos { get; set; }
        public int Vitorias { get; set; }
        public int Empates { get; set; }
        public int Derrotas { get; set; }
        public int GolsPro { get; set; }
        public int GolsContra { get; set; }
        public int SaldoGols { get; set; }
        public int Pontos { get; set; }
        public double Aproveitamento { get; set; }
    }

    public class MediaPosicaoDto
    {
        public string Posicao { get; set; } = string.Empty;
        public double Media { get; set; }
        public int TotalJogadores { get; set; }
    }
}
